app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,97 @@
 class SendgridApi
-  extend SingleForwardable
+  TIMEOUT = 2 # seconds
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  RESCUABLE_ERRORS = [SendgridClient::Error,
+                      JSON::ParserError,
+                      Excon::Errors::Error,
+                      Timeout::Error
+                     ].freeze
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    action = 'spamreports.get.json'
+    query = { email: email }
+
+    response = SendgridPool.instance.with { |conn|
+      conn.get_request(action, query)
+    }
+
+    return false if error_response?(response)
+
+    response.any?
+  rescue *RESCUABLE_ERRORS => e
+    Rails.logger.error("#{e.class.name}: #{e.message}")
+    false
   end
 
   def bounced?(email)
-    api { bounces.retrieve(email: email).any? }
+    action = 'bounces.get.json'
+    query = { email: email }
+
+    response = SendgridPool.instance.with { |conn|
+      conn.get_request(action, query)
+    }
+
+    return false if error_response?(response)
+
+    response.any?
+  rescue *RESCUABLE_ERRORS => e
+    Rails.logger.error("#{e.class.name}: #{e.message}")
+    false
   end
 
   def remove_from_bounce_list(email)
-    api { bounces.delete(email: email) }
+    action = 'bounces.delete.json'
+    data = { email: email }
+
+    response = SendgridPool.instance.with { |conn|
+      conn.post_request(action, data)
+    }
+
+    return false if error_response?(response)
+
+    email_removed?(response)
+  rescue *RESCUABLE_ERRORS => e
+    Rails.logger.error("#{e.class.name}: #{e.message}")
+    false
   end
 
   def remove_from_spam_list(email)
-    api { spam_reports.delete(email: email) }
-  end
+    action = 'spamreports.delete.json'
+    data = { email: email }
 
-private
+    response = SendgridPool.instance.with { |conn|
+      conn.post_request(action, data)
+    }
+
+    return false if error_response?(response)
 
-  def api(&_action)
-    yield
-  rescue => e
-    Rails.logger.error("SendgridApi error: #{e.class} #{e}")
+    email_removed?(response)
+  rescue *RESCUABLE_ERRORS => e
+    Rails.logger.error("#{e.class.name}: #{e.message}")
     false
   end
 
-  def spam_reports
-    SendgridToolkit::SpamReports.new
+private
+
+  def error_response?(response)
+    # Response could be empty which gets translated to an Array or have data
+    # which becomes a Hash according to the specs
+    if response.try(:key?, 'error')
+      msg = "'#{self.class.name}' sendgrid response: #{response['error']}"
+      Rails.logger.error(msg)
+      true
+    else
+      false
+    end
   end
 
-  def bounces
-    SendgridToolkit::Bounces.new
+  def email_removed?(response)
+    if response.try(:key?, 'message') && response['message'] != 'success'
```

Comment:
Testing the negative condition feels awkward, and it misses cases (e.g. if there is no message at all). If we're going to rescue all errors I would suggest that we write this code with confidence about the expected types, so simply:

    if response['message'] == 'success'